roles/jd-server/src/lib/job_declarator/message_handler.rs

```diff
@@ -38,6 +41,41 @@ impl JobDeclaratorDownstream {
     }
 }
 
+pub fn clear_old_mempool_transactions(
```

Comment:
Could we compare old tx list and new tx list and only remove whats out then? that could be more efficient if there is enough similarity between the mining jobs

Reply:
@lorbax  @GitGab19 what do you think? This looks like a good idea, it would mean there would be no need for the retain function in the update_mempool function

Reply:
What new tx list are you talking about? And which old tx list? Are you referring to node mempool or jds mempool?

Reply:
If you're referring to mempool management (it's done in the other part of this PR), I think that @NonsoAmadi10 is already trying to do what you are mentioning here (maybe I'm missing something)

Reply:
> Could we compare old tx list and new tx list and only remove whats out then? that could be more efficient if there is enough similarity between the mining jobs

yes would make a lot of sense actually, but this is supposed to be a kind of first issue for the author. Perhaps to keep it like it is and open an issue? What do you think?